lib/simplifyExpression/basicsSearch/simplifyDoubleUnaryMinus.js

```diff
@@ -2,36 +2,28 @@ const clone = require('../../util/clone');
 
 const ChangeTypes = require('../../ChangeTypes');
 const Node = require('../../node');
+const {build, query} = require('math-nodes');
 
 // Simplifies two unary minuses in a row by removing both of them.
 // e.g. -(- 4) --> 4
 function simplifyDoubleUnaryMinus(node) {
-  if (!Node.Type.isUnaryMinus(node)) {
+  if (!query.isNeg(node)) {
     return Node.Status.noChange(node);
   }
   const unaryArg = node.args[0];
+  const val = query.getValue(unaryArg);
   // e.g. in - -x, -x is the unary arg, and we'd want to reduce to just x
-  if (Node.Type.isUnaryMinus(unaryArg)) {
+  if (query.isNeg(unaryArg)) {
     const newNode = clone(unaryArg.args[0]);
     return Node.Status.nodeChanged(
       ChangeTypes.RESOLVE_DOUBLE_MINUS, node, newNode);
   }
   // e.g. - -4, -4 could be a constant with negative value
-  else if (Node.Type.isConstant(unaryArg) && parseFloat(unaryArg.value) < 0) {
-    const newNode = Node.Creator.constant(parseFloat(unaryArg.value) * -1);
+  else if (query.isNumber(unaryArg) && parseFloat(val) < 0) {
+    const newNode = build.number(parseFloat(val) * -1);
     return Node.Status.nodeChanged(
       ChangeTypes.RESOLVE_DOUBLE_MINUS, node, newNode);
   }
-  // e.g. -(-(5+2))
-  else if (Node.Type.isParenthesis(unaryArg)) {
```

Comment:
The only time the new parser emits a `parentheses` node is when they aren't necessary, e.g `1 + (2)`.  We remove all unnecessary parentheses from user input before proceeding so we should never run into a `parentheses` node here.  In this case `-(-(5+2))` is represented as follows:
```
{
   type: 'Apply',
   op: 'neg',
   args: [{
      type: 'Apply',
      op: 'neg',
      args: [{
         type: 'Apply',
         op: 'add',
         args: [{type: 'Number', value: '5'}, {type: 'Number', value: '2'}],
      }],
   }],
}
```
If you're curious about what different input looks like, there's a demo of the parser at https://semantic-math.github.io/math-parser/.